app/controllers/communicarts_controller.rb

```diff
@@ -21,9 +21,16 @@ def send_cart
   def approval_response
     cart = Cart.find(params[:cart_id]).decorate
     client_data = cart.proposal.client_data_legacy
-    approval = cart.approvals.find_by(user_id: user_id)
+    approval = cart.approvals.approvable.find_by(user_id: user_id)
+    
+    if !approval
+      flash[:error] = "Sorry. You are not allowed to approve your own request."
+      redirect_to cart_path(cart)
+      return 
```

Comment:
I'm personally allergic to mid-method `return`s...mind using an `if`/`else` instead? @cmc333333 Curious to hear your thoughts.